src/load/__tests__/decodeBmp.test.ts

```diff
@@ -0,0 +1,75 @@
+import { expect, test } from 'vitest';
+
+import { decodeBmp } from '../decodeBmp.js';
+
+test('should decode RGBA image', () => {
+  const result = decodeBmp(testUtils.loadBuffer('formats/bmp/2x2RGBA.bmp'));
+
+  expect(result.channels).toBe(4);
+  expect(result.components).toBe(3);
+  expect(result.getRawImage().data).toStrictEqual(
+    new Uint8Array(
+      [
+        [255, 0, 0, 255, 0, 255, 0, 255],
+        [0, 0, 255, 255, 255, 255, 255, 255],
+      ].flat(),
+    ),
+  );
+});
```

Comment:
Where did you get this test pattern from? We have various helpers to create images for test assertions to avoid complicated code like this.